File: frontend/src/types/schema.ts
/**
 * Type definitions for schema operations
 */

export interface ColumnSchema {
  id: number;
  column_name: string;
  column_index: number;
  data_type: string;
  sample_values: any[] | null;
  null_count: number;
  unique_count: number;
  statistics: ColumnStatistics | null;
}

export interface ColumnStatistics {
  min: number;
  max: number;
  mean: number;
  median: number;
  std: number;
}

export interface FileSchemaResponse {
  file_id: number;
  filename: string;
  row_count: number;
  column_count: number;
  encoding: string | null;
  columns: ColumnSchema[];
}

export interface SchemaAnalysisRequest {
  sample_size?: number;
}

File: backend/app/schemas/schema.py
"""
Pydantic schemas for schema operations.
"""
from typing import List, Optional, Dict, Any
from pydantic import BaseModel


class ColumnSchema(BaseModel):
    """Schema for a single column."""
    
    id: int
    column_name: str
    column_index: int
    data_type: str
    sample_values: Optional[List[Any]] = None
    null_count: int
    unique_count: int
    statistics: Optional[Dict[str, Any]] = None
    
    class Config:
        from_attributes = True


class FileSchemaResponse(BaseModel):
    """Schema response for a file with all columns."""
    
    file_id: int
    filename: str
    row_count: int
    column_count: int
    encoding: Optional[str]
    columns: List[ColumnSchema]


class SchemaAnalysisRequest(BaseModel):
    """Request to trigger schema analysis."""
    
    sample_size: Optional[int] = 5  # Number of sample values to extract
